packages/flutter/lib/src/rendering/sliver_persistent_header.dart

```diff
@@ -76,6 +109,23 @@ abstract class RenderSliverPersistentHeader extends RenderSliver with RenderObje
   double _lastShrinkOffset = 0.0;
   bool _lastOverlapsContent = false;
 
+  /// Defines the parameters used to execute a [StretchCallback] when a
+  /// stretching header over-scrolls.
+  ///
+  /// If [stretchConfiguration] is null then callback is not triggered.
+  ///
+  /// See also:
+  ///
+  ///  * [SliverAppBar], which creates a header that can stretched into an
+  ///  overscroll area and trigger a callback function.
+  OverScrollHeaderStretchConfiguration get stretchConfiguration => _stretchConfiguration;
+  OverScrollHeaderStretchConfiguration _stretchConfiguration;
+  set stretchConfiguration(OverScrollHeaderStretchConfiguration value) {
+    if (value == _stretchConfiguration)
+      return;
+    _stretchConfiguration = value;
+  }
```

Comment:
I think I've seen this pattern in our code base a lot, so this is probably just me not knowing enough about it, but why do all this getter/setter work and not just use a public attribute here?

Reply:
I did not see anything in the style guide beyond using this construct to assert conditions in the setter. I was just following conventions I already saw in place. 😅 I think I can remove it then?

Reply:
Ah that would make sense for assertions.  To me it makes sense to remove it since there is no assertion in this case, but up to you.